x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/detail/settings/install_button.tsx

```diff
@@ -13,23 +13,37 @@ import type { PackageInfo, UpgradePackagePolicyDryRunResponse } from '../../../.
 import { InstallStatus } from '../../../../../types';
 import { useAuthz, useGetPackageInstallStatus, useInstallPackage } from '../../../../../hooks';
 
+import { getNumTransformAssets } from '../../../../../../../components/transform_install_as_current_user_callout';
+
 import { ConfirmPackageInstall } from './confirm_package_install';
-type InstallationButtonProps = Pick<PackageInfo, 'name' | 'title' | 'version'> & {
+
+type InstallationButtonProps = Pick<PackageInfo, 'name' | 'title' | 'version' | 'assets'> & {
   disabled?: boolean;
   dryRunData?: UpgradePackagePolicyDryRunResponse | null;
   isUpgradingPackagePolicies?: boolean;
   latestVersion?: string;
-  numOfAssets: number;
   packagePolicyIds?: string[];
   setIsUpgradingPackagePolicies?: React.Dispatch<React.SetStateAction<boolean>>;
 };
 export function InstallButton(props: InstallationButtonProps) {
-  const { name, numOfAssets, title, version } = props;
+  const { name, title, version, assets } = props;
+
   const canInstallPackages = useAuthz().integrations.installPackages;
   const installPackage = useInstallPackage();
   const getPackageInstallStatus = useGetPackageInstallStatus();
   const { status: installationStatus } = getPackageInstallStatus(name);
 
+  const numOfAssets = Object.entries(assets).reduce(
```

Comment:
are all assets tranform type here?

Reply:
I'm not sure what you are asking? `numOfAssets` is getting a count of all assets, and `numOfTransformAssets` gets the subset count of `transform` type assets.

`numOfTransformAssets` is used both here and on the policy editor page you linked to in order to display an info callout about transforms being created with the current user's role (if number of transforms > 0). here this callout is displayed in the confirm install modal. on the policy editor page it is displayed inline.

I pulled both of them into this install button component since this component is now used in multiple places, to avoid having to pass it down from different parent components

Reply:
My bad, for some reason I thought `numOfAssets` is passed to `getNumTransformAssets`, never mind.